### PR TITLE
Improve related image results on photo detail page

### LINKED ISSUE #123: Improve related image results on photo detail page
It looks like related images are generated from a search of the first tag in the image. At a minimum, this should reuse the user's search query instead of a random tag.

We could also off-load this functionality to the backend and develop an /image/{id}/related endpoint if we need something more sophisticated than that. I'll bring this up during our standup on Monday.

### PR DESCRIPTION
**Pull request by VatsalSin**
-> Related Image section now shows images related to search query posed by the user rather than using the first tag.
-> While testing sometimes the stored query returned null, hence when the this.$store.state.query will return null, related images will show results according to earlier criteria (based on first image tag). 

Fixes #123

### CODE REVIEW

**Comment by VatsalSin** [comment]
I will add the tests at the earliest.

**Comment by VatsalSin** [comment]
@brenoferreira I have added test for thr RelatedImages component. Please review the changes.

File: src/pages/PhotoDetailPage.vue [hunk 1]
```diff
@@ -110,15 +110,20 @@ const PhotoDetailPage = {
       this.isPrimaryImageLoaded = true;
     },
     getRelatedImages(tags, query) {
-      let queryParam = query;
-      const tagsParam = (tags || []).slice();
-
-      if (tagsParam.length > 0) {
-        queryParam = tagsParam.slice(0, 1).map(tag => tag.name).join(', ');
+      if (this.query.q) {
+        console.log(this.$store.state.query.q);
```
Comment (brenoferreira) [inline src/pages/PhotoDetailPage.vue]: Could you remove this `console.log` call please?

**Comment by VatsalSin** [comment]
@brenoferreira I have made the required changes and resolved the merge conflicts. Please review the changes.

**Comment by brenoferreira** [comment]
So, we decided to go with this. Thank you for your contribution @VatsalSin 👍

**Pull request merged**